applications/PfemFluidDynamicsApplication/custom_processes/select_mesh_elements_for_fluids_process.hpp

```diff
@@ -47,589 +47,455 @@ namespace Kratos
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// This operator is provided to call the process as a function and simply calls the Execute method.
+    void operator()()
+    {
+        Execute();
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Execute method is used to execute the Process algorithms.
+    void Execute() override
+    {
+        KRATOS_TRY
+
+        if (mEchoLevel > 1) {
+            std::cout << " [ SELECT MESH ELEMENTS in PfemFluid: (" << mrRemesh.OutMesh.GetNumberOfElements() << ") " << std::endl;
+            std::cout << "MODEL PART InNumberOfElements " << mrRemesh.InMesh.GetNumberOfElements() << std::endl;
+            std::cout << "MODEL PART InNumberOfPoints " << mrRemesh.InMesh.GetNumberOfPoints() << std::endl;
+            std::cout << "MODEL PART OutNumberOfElements " << mrRemesh.OutMesh.GetNumberOfElements() << std::endl;
+            std::cout << "MODEL PART OutNumberOfPoints " << mrRemesh.OutMesh.GetNumberOfPoints() << std::endl;
+        }
+        const int &OutNumberOfElements = mrRemesh.OutMesh.GetNumberOfElements();
+        mrRemesh.PreservedElements.clear();
+        mrRemesh.PreservedElements.resize(OutNumberOfElements);
+        std::fill(mrRemesh.PreservedElements.begin(), mrRemesh.PreservedElements.end(), 0);
+        mrRemesh.MeshElementsSelectedFlag = true;
+
+        mrRemesh.Info->NumberOfElements = 0;
+
+        const ProcessInfo &rCurrentProcessInfo = mrModelPart.GetProcessInfo();
+        double currentTime = rCurrentProcessInfo[TIME];
+        double timeInterval = rCurrentProcessInfo[DELTA_TIME];
+        bool firstMesh = false;
+        if (currentTime < 2 * timeInterval) {
+            firstMesh = true;
+        }
+
+        bool box_side_element = false;
+        bool wrong_added_node = false;
+
+        int number_of_slivers = 0;
+
+        if (mrRemesh.ExecutionOptions.IsNot(MesherUtilities::SELECT_TESSELLATION_ELEMENTS)) {
+            for (int el = 0; el < OutNumberOfElements; el++) {
+                mrRemesh.PreservedElements[el] = 1;
+                mrRemesh.Info->NumberOfElements += 1;
+            }
+        } else {
+            if (mEchoLevel > 1)
+                std::cout << " Start Element Selection " << OutNumberOfElements << std::endl;
+
+            ModelPart::ElementsContainerType::iterator element_begin = mrModelPart.ElementsBegin();
+            const unsigned int nds = element_begin->GetGeometry().size();
+            const unsigned int dimension = element_begin->GetGeometry().WorkingSpaceDimension();
+            int *OutElementList = mrRemesh.OutMesh.GetElementList();
+            ModelPart::NodesContainerType &rNodes = mrModelPart.Nodes();
+
+            int el = 0;
+            int number = 0;
+
+            //#pragma omp parallel for reduction(+:number) private(el)
+            for (el = 0; el < OutNumberOfElements; el++) {
+                Geometry<Node<3>> vertices;
+
+                unsigned int numfreesurf = 0;
```

Comment:
This is not efficient, it is not openmp and you are allocating inside the loop

Reply:
@AFranci maybe we should check this... This process is called at each step so maybe we can reduce a lot of time